napari_plugin_manager/npe2api.py

```diff
@@ -0,0 +1,123 @@
+"""
+These convenience functions will be useful for searching pypi for packages
+that match the plugin naming convention, and retrieving related metadata.
+"""
+
+import json
+from collections.abc import Iterator
+from concurrent.futures import ThreadPoolExecutor
+from functools import lru_cache
+from typing import (
+    Optional,
+    TypedDict,
+    cast,
+)
+from urllib.error import HTTPError, URLError
+from urllib.request import Request, urlopen
+
+from napari.plugins.utils import normalized_name
+from napari.utils.notifications import show_warning
+from npe2 import PackageMetadata
+from typing_extensions import NotRequired
+
+PyPIname = str
```

Comment:
```suggestion
PyPIName = str
```

I'd normally type it like that but it looks weird :/